linetools/isgm/utils.py

```diff
@@ -463,6 +471,127 @@ def group_coincident_compoments(comp_list, output_type='list'):
+    lst=np.array(lst)
+    compnos=np.array(compnos)
+    wv1s=np.array([line.limits.wvlim[0].value for line in lst])
+    sortidxs=np.argsort(wv1s)
+    sort_lst=lst[sortidxs]
+    sort_compnos=compnos[sortidxs] # This will store indices of the lines' parent comps.
+
+    ### Identify the blends
+    blends = []
+    for i in range(len(sortidxs)-1):
+        if i == 0:
+            thisblend = [i]
+        if sort_lst[i].coincident_line(sort_lst[i+1]):
+            thisblend.append(i+1)
+            if i==(len(sortidxs)-2):
+            	blends.append(thisblend)
+        else:
+            blends.append(thisblend)
+            thisblend = [i+1]
+            if i==(len(sortidxs)-2):
+            	blends.append(thisblend)
+
+    ### Associate the lines to their parent components
+    blendnos=[]
+    for blist in blends:
+        blendnos.append(sort_compnos[blist])
+
+    ### Main algorithm to group together all components with blended lines
+    compfound=[]
+    grblends=[]
+    newgroups=[]
+    for i,bn in enumerate(blendnos):
+        if i in grblends: continue
+        grblends.append(i)
+        newgroups.append(bn.tolist())
+        newtotry=bn
+        while (len(newtotry)>0):
+            newnewtotry=[]
+            for no in newtotry:
+                if no not in compfound:
+
+                    compfound.append(no)
+                    blgrs=wherein(blendnos,no)
+                    for bg in blgrs:
+                        if bg not in grblends:
+                            grblends.append(bg)
+                            newgroups[-1].extend(blendnos[bg].tolist())
+                            newnewtotry.extend(np.unique(blendnos[bg]).tolist())
+            newtotry=newnewtotry
+            newgroups[-1]=np.unique(np.array(newgroups[-1])).tolist()
+
+    out = newgroups
+
+    # Now lets produce the final output from it
+    output_list = []
+    output_dict = {}
+    for ii in range(len(out)):
+        aux_list = []
+        for jj in out[ii]:
+            aux_list += [comp_list[jj]]
+        output_dict['{}'.format(ii)] = aux_list
+        output_list += [aux_list]
+
+    # choose between dict of list
+    if output_type == 'list':
+        return output_list
+    elif output_type in ['dict', 'dictionary']:
+        return output_dict
+
+def wherein(groups,member):
+    ''' Once blends have been identified, find which blend a given line index belongs to.
+    Parameters
+    ----------
+    groups : list of lists
+        Input list of groups within which you want to find a member
+        Here, this is used for groups of blended line indices
+    member : value
+        Value for which to search within groups.
+        Here, this corresponds to a specific line index
+
+    Returns
+    -------
+    matches : list
+        Indices of groups within with 'member' was found
+    '''
+    matches=[]
+    for i,gr in enumerate(groups):
+        if member in gr:
+            matches.append(i)
+    return matches
```

Comment:
This function seems very generic; I wonder whether there may be something like this already implemented in python lists.